hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsS3Factory.kt

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.hmpps.sqs
+
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.runBlocking
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.context.ConfigurableApplicationContext
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest
+import software.amazon.awssdk.services.s3.model.ListBucketsRequest
+
+class HmppsS3Factory(
+  private val context: ConfigurableApplicationContext,
+  private val s3ClientFactory: S3ClientFactory,
+) {
```

Comment:
This PR is perfect for what it's doing... my only concern is that it introduces another AWS service. So now it's not the SQS lib, it's SQS + S3 lib. Possibly better if each AWS service has its own lib? Or maybe I'm overly opbsessed with small apps and libraries 😆. Let's have a chat. 😄